### PR TITLE
Bugfix/wrong size attributes in image with border

### LINKED ISSUE #30186: [Issue] Bugfix/wrong size attributes in image with border
This issue is automatically created based on existing pull request: magento/magento2#30139: Bugfix/wrong size attributes in image with border

---------

### Summary (*)
Magento 2.4-develop
This PR removes most changes from commit [#4e085c7](https://github.com/magento/magento2/commit/ad24cb8cea4c98fb074510e33fc9ee514cb0ba58)

Commit [#4e085c7](https://github.com/magento/magento2/commit/ad24cb8cea4c98fb074510e33fc9ee514cb0ba58) tried to fix images getting distorted.
By setting a boolean for the `image_with_border.phtml`.
This boolean would set the width and height attribute, to the invalid html max-width and max-height attribute.
While this did fix the images not getting distorted.
This did reintroduced the use of invalid html attributes.

This commit unsets the boolean, use of invalid html attributes and fixes the images getting distorted, via the CSS instead.

### Proposed solution

1. Remove the boolean `enable_lazy_loading_for_images_without_borders`
2. Remove the max-width and max-height and the if statement holding this.
3. Add the value `width: auto` to the `.product-image-photo` (This also possible via `object-fit: scale-down`)

_This PR also removes the redundant max-width and height from the `.product-image-photo`._
_Since these values are already set globally, on the image, via the reset styles._

### PR DESCRIPTION
**Pull request by GrimLink**
### Description (*)
This PR removes most changes from commit [#4e085c7](https://github.com/magento/magento2/commit/ad24cb8cea4c98fb074510e33fc9ee514cb0ba58)

Commit [#4e085c7](https://github.com/magento/magento2/commit/ad24cb8cea4c98fb074510e33fc9ee514cb0ba58) tried to fix images getting distorted.
By setting a boolean for the `image_with_border.phtml`.
This boolean would set the width and height attribute, to the invalid html max-width and max-height attribute.
While this did fix the images not getting distorted.
This did reintroduced the use of invalid html attributes.

This commit unsets the boolean, use of invalid html attributes and fixes the images getting distorted, via the CSS instead.

### Manual testing scenarios (*)

1. Remove the boolean `enable_lazy_loading_for_images_without_borders`
2. Remove the max-width and max-height and the if statement holding this.
3. Add the value `width: auto` to the `.product-image-photo` (This also possible via `object-fit: scale-down`)

_This PR also removes the redundant max-width and height from the `.product-image-photo`._
_Since these values are already set globally, on the image, via the reset styles._

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#30186: Bugfix/wrong size attributes in image with border

### CODE REVIEW

**Comment by GrimLink** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
Can you verify previous commit purpose related with this doc https://devdocs.magento.com/guides/v2.4/frontend-dev-guide/themes/theme-images.html. Seem they try to add new variable for solve problem  
> This is done to prevent small images from being stretched and distorted.

Also i think doc should be update for clarify user to use correct image size.

**Comment by GrimLink** [comment]
@mrtuvn I can confirm that the previous commit is related to the doc https://devdocs.magento.com/guides/v2.4/frontend-dev-guide/themes/theme-images.html#lazy_loading

But I don't understand why the previous commit was seen as valid 😖

This same issue also plagued the images in the minicart.
And those images did not have the native lazy loading on it.
This was also fixed by unsetting the width and height on the image size.

The width and height are recommended for the native lazy loading and to prevent layout jank.
But unsetting them is not wrong.
As the images are rendered in a aspect-ratio that all ready reserves the image space.

But I will check how to also update the doc with this changed.
Thank for letting me know of this doc section 😉

File: app/design/frontend/Magento/blank/Magento_Catalog/web/css/source/_module.less [hunk 1]
```diff
@@ -115,15 +115,14 @@
...
-        height: auto;
         left: 0;
+        display: block;
+        width: auto;
         margin: auto;
-        max-width: 100%;
-        position: absolute;
-        right: 0;
-        top: 0;
     }
```
Comment (mrtuvn) [inline app/design/frontend/Magento/blank/Magento_Catalog/web/css/source/_module.less:127]: you should sort styles properties for both file less changed

Comment (GrimLink) [reply app/design/frontend/Magento/blank/Magento_Catalog/web/css/source/_module.less:127]: I did sort it for both files, what did I mis?

Comment (mrtuvn) [reply app/design/frontend/Magento/blank/Magento_Catalog/web/css/source/_module.less:127]: magento require all properties in less should sort alphabetical. You can see fail static tests report for more details

**Comment by mrtuvn** [comment]
@GrimLink pull request prefix MC- meaning this made by internal core team so you probably will not see issue or pull request open. I'm not sure more details about it . You can ping author here for asking 
author related PR: @adifucan
CC: @krzksz can you review this

**Review by krzksz — CHANGES_REQUESTED** [review]
If I understand this correctly and previous change was some hacky workaround that we can remove while fixing the issue with those few lines of CSS then I'm happy to approve this.

Let's just resolve those failing tests that @mrtuvn mentioned.

**Comment by krzksz** [comment]
@magento run Functional Tests B2B,Functional Tests CE,Functional Tests EE,Static Tests

File: app/design/frontend/Magento/blank/Magento_Catalog/web/css/source/_module.less (hunk 1 above)
Comment (GrimLink) [reply app/design/frontend/Magento/blank/Magento_Catalog/web/css/source/_module.less:127]: okay fixed it.

**Comment by krzksz** [comment]
@magento run Static Tests

**Comment by GrimLink** [comment]
@krzksz I am not sure how to als update the docs https://devdocs.magento.com/guides/v2.4/frontend-dev-guide/themes/theme-images.html#lazy_loading
Can you point me in the right direction on how to do this?

**Comment by krzksz** [comment]
@magento run all tests

**Comment by krzksz** [comment]
@GrimLink Hey, you can find the repository here: https://github.com/magento/devdocs

When making PR there you should be able to reference this one to make sure they are processed together.

**Comment by krzksz** [comment]
@magento run Functional Tests B2B,WebAPI Tests

**Review by adifucan — APPROVED** [review]
Thanks @GrimLink! Tested it locally. Nice job!
Yes, please, update https://devdocs.magento.com/guides/v2.4/frontend-dev-guide/themes/theme-images.html#lazy_loading as well.

**Review by mrtuvn — APPROVED** [review]

**Review by krzksz — APPROVED** [review]

**Comment by krzksz** [comment]
I don't think it is possible to cover this scenario with automatic tests.

**Comment by magento-engcom-team** [comment]
Hi @krzksz, thank you for the review.
[ENGCOM-8251](https://jira.corp.magento.com/browse/ENGCOM-8251) has been created to process this Pull Request
:eight_spoked_asterisk: @krzksz, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed
**Result:**
:heavy_check_mark: Small images are not stretched/distorted and displayed as before
image
image

**Comment by engcom-Delta** [comment]
@magento create issue

**Comment by engcom-Delta** [comment]
**Note:** Automation tests are passed

**Comment by gabrieldagama** [comment]
@magento give me new instance with edition ee

**Pull request merged**